tensorizer/serialization.py

```diff
@@ -2858,19 +2919,24 @@ def _copy_thread(
                 if halt:
                     break
 
-                header = _TensorHeaderDeserializer.from_io(
-                    file_,
-                    zero_hashes=True,
-                    check_crypt_info=unsafe_self._has_crypt_info,
-                )
-
-                if header is None:
-                    raise ValueError("Unexpected empty header")
+                if unsafe_self._headers is None:
+                    header = _TensorHeaderDeserializer.from_io(
+                        unsafe_self._file_header.version_number,
+                        file_,
+                        zero_hashes=True,
+                        check_crypt_info=unsafe_self._has_crypt_info,
+                    )
+                    if header is None:
+                        raise KeyError("Unexpected empty header")
```

Comment:
Why is this a `KeyError` instead of a `ValueError`?

Reply:
Updated!